### DISCUSSION TITLE
Breaking Change on Version 1?

### DISCUSSION THREAD

**Discussion by jacklimwenjie** [vercel/swr#1300] [General] [answered]
As title, would there be any breaking change on version 1?

**Comment by shuding** [comment]
For now, the only breaking API change in v1 will be the import path for `SWRInfinite`:

Old: `import { useSWRInfinite } from 'swr'`
New: `import useSWRInfinite from 'swr/infinite'`

If you're using TypeScript, there're also some deprecated types that will be renamed in v1 (#1276, we've added the deprecation message months ago):
- `ConfigInterface` will be renamed to `SWRConfiguration`
- `keyInterface` will be renamed to `Key`
- `responseInterface` will be renamed to `SWRResponse`
- `SWRInfiniteConfigInterface` will be renamed to `SWRInfiniteConfiguration`
- `SWRInfiniteConfiguration` will be moved to `swr/infinite`
- `SWRInfiniteResponseInterface` will be renamed to `SWRInfiniteResponse` in `swr/infinite`
- `SWRInfiniteResponse` will be moved to `swr/infinite`
- `RevalidateOptionInterface` will be renamed to `RevalidatorOptions`
- `revalidateType` will be renamed to `Revalidator`

---

Update: there are another 2 breaking changes:
- `useSWR` will not return the `revalidate` function anymore, call the returned bound mutate function `mutate()` without any argument for the same purpose
- `useSWR` will not provide a default `fetcher` anymore since #1304.

**Reply by nandorojo** [reply]
Also, `cache` is no longer exported from `'swr'`

**Reply by shuding** [reply]
> Also, `cache` is no longer exported from `'swr'`

Yep, the `cache` API wasn't documented before as we didn't recommend people to use it. We should have put a "unstable_" prefix to it.

**Reply by rodbs** [reply]
But is it possible to use `cache ` then? Where should we export it now in version 1.0.0? is it documented anywhere? thx

**Reply by shuding** [reply]
Yeah it's still possible to get access to `cache`. We will release the 1.0 docs soon!

**Reply by shuding** [reply]
Hi all! You can find the announcement and migration guide for 1.0 here: https://swr.vercel.app/blog/swr-v1

**Comment by shuding** [comment, marked as answer]
FYI our blog post and migration guide for 1.0 is out: https://swr.vercel.app/blog/swr-v1 🎉
